bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/osgi/ResourceBundleClassLoader.java

```diff
@@ -172,15 +187,26 @@ private Charset getResourceCharset(URL url) {
         return charset;
     }
 
+    /**
+     * The method tests whether the file content can be decoded with the given character set.
+     *
+     * @param f
+     *            name of the file that will be tested
+     * @param charset
+     *            character set to which the file is tested
+     * @return
+     *         passed character set if the file could be decoded with this character set or null if the decode fails
+     */
     private Charset detectCharset(String f, Charset charset) {
         try {
             InputStream in = super.getResourceAsStream(f);
             BufferedInputStream input = new BufferedInputStream(in);
             CharsetDecoder decoder = charset.newDecoder();
             decoder.reset();
             byte[] buffer = new byte[512];
-            boolean identified = false;
-            while ((input.read(buffer) != -1) && (!identified)) {
```

Comment:
@NorbertHD I have tested the first two cases (and also added tests to the new unit test class). About the last case you mentioned I am not absolutely sure. I will test it.